client/src/cmdhficlass.c

```diff
@@ -5403,18 +5404,66 @@ static int CmdHFiClassSAM(const char *Cmd) {
     void *argtable[] = {
         arg_param_begin,
         arg_lit0("v", "verbose", "verbose output"),
+        arg_lit0("k", "keep", "keep the field active after command executed"),
+        arg_lit0("n", "nodetect", "skip selecting the card and sending card details to SAM"),
+        arg_lit0("t",  "tlv",      "decode TLV"),
+        arg_lit0(NULL, "break-on-nr-mac", "stop tag interaction on nr-mac"),
+        arg_lit0("p", "prevent-epurse-update", "fake epurse update"),
+        arg_lit0(NULL, "shallow", "shallow mod"),
+        arg_strx0("d", "data",     "<hex>", "DER encoded command to send to SAM"),
         arg_param_end
     };
     CLIExecWithReturn(ctx, Cmd, argtable, true);
-    bool verbose = arg_get_lit(ctx, 1);
+
+    bool verbose = false;
+    if (arg_get_lit(ctx, 1)) {
+        verbose = true;
+    }
+    bool disconnectAfter = true;
+    if (arg_get_lit(ctx, 2)) {
+        disconnectAfter = false;
+    }
+    bool skipDetect = false;
+    if (arg_get_lit(ctx, 3)) {
+        skipDetect = true;
+    }
+    bool decodeTLV = false;
+    if (arg_get_lit(ctx, 4)) {
+        decodeTLV = true;
+    }
+    bool breakOnNrMac = false;
+    if (arg_get_lit(ctx, 5)) {
+        breakOnNrMac = true;
+    }
+    bool preventEpurseUpdate = false;
+    if (arg_get_lit(ctx, 6)) {
+        preventEpurseUpdate = true;
+    }
+    bool shallow_mod = false;
+    if (arg_get_lit(ctx, 7)) {
+        shallow_mod = true;
+    }
+
+    uint64_t command_flags = 
+        (disconnectAfter ? 1 : 0) +
+        (skipDetect ? 2 : 0) +
+        (breakOnNrMac ? 4 : 0) +
+        (preventEpurseUpdate ? 8 : 0) +
+        (shallow_mod ? 16 : 0);
+
+
+    uint8_t data[PM3_CMD_DATA_SIZE] = {0};
+    int datalen = 0;
+    CLIGetHexBLessWithReturn(ctx, 8, data, &datalen, 0);
+
     CLIParserFree(ctx);
 
     if (IsHIDSamPresent(verbose) == false) {
         return PM3_ESOFT;
     }
 
     clearCommandBuffer();
-    SendCommandNG(CMD_HF_SAM_PICOPASS, NULL, 0);
+    SendCommandMIX(CMD_HF_SAM_PICOPASS, command_flags, 0, datalen, data, datalen);
```

Comment:
Please rework to use structs and the SendCommandNG() call.   MIX is to be considered deprecated.

Reply:
Changed here and in SEOS.